unified_planning/io/anml_writer.py

```diff
@@ -0,0 +1,329 @@
+
+        converter = ConverterToANMLString(self.problem.env)
+
+        for a in self.problem.actions:
+            if isinstance(a, up.model.InstantaneousAction):
+                parameters = [f'{_get_mangled_type_name(ap.type, types_mapping)} {ap.name}' for ap in a.parameters]
+                out.write(f'action {a.name}({", ".join(parameters)}) {{\n')
+                for p in a.preconditions:
+                    out.write(f'   [ start ] {converter.convert(p)};\n')
+                for e in a.effects:
+                    out.write(f'   {_convert_effect(e, converter)}')
+                out.write('}\n')
+            elif isinstance(a, DurativeAction):
+                parameters = [f'{_get_mangled_type_name(ap.type, types_mapping)} {ap.name}' for ap in a.parameters]
+                out.write(f'action {a.name}({", ".join(parameters)}) {{\n')
+                out.write
+                for i, cl in a.conditions.items():
+                    for c in cl:
+                        out.write(f'   {_convert_anml_interval(i)} {converter.convert(c)};\n')
+                for ti, el in a.effects.items():
+                    for e in el:
+                        out.write(f'   {_convert_effect(e, converter, ti)}')
+                out.write('}\n')
+            else:
+                raise NotImplementedError
+
+        for t in self.problem.user_types: # Define objects
+            objects: List['unified_planning.model.Object'] = list(self.problem.objects(t))
+            obj_names = [o.name for o in self.problem.objects(t)]
+            if len(objects) > 0:
+                out.write(f'instance {_get_mangled_type_name(t, types_mapping)} {", ".join(obj_names)};\n')
+
+        for fe, v in self.problem.initial_values.items():
+            out.write(f'[ start ] {converter.convert(fe)} := {converter.convert(v)};\n')
+
+        for ti, el in self.problem.timed_effects.items():
+            for e in el:
+                out.write(_convert_effect(e, converter, ti))
+
+        for g in self.problem.goals:
+            out.write(f'[ end ] {converter.convert(g)};\n')
+
+        for i, gl in self.problem.timed_goals.items():
+            for g in gl:
+                out.write(f'{_convert_anml_interval(i)} {converter.convert(g)};\n')
+
+    def print_problem(self):
+        '''Prints to std output the ANML problem.'''
+        self._write_problem(sys.stdout)
+
+    def get_problem(self) -> str:
+        '''Returns the ANML problem.'''
+        out = StringIO()
+        self._write_problem(out)
+        return out.getvalue()
+
+    def write_problem(self, filename: str):
+        '''Dumps to file the ANML problem.'''
+        with open(filename, 'w') as f:
+            self._write_problem(f)
+
+def _convert_effect(effect: 'up.model.Effect', converter: ConverterToANMLString, timing: 'up.model.Timing' = None) -> str:
+    results: List[str] = []
+    if effect.is_conditional():
+        results.append(f'when [ all ] {converter.convert(effect.condition)}\n{{')
```

Comment:
I think the correct way to convert conditional effects is:
```suggestion
    if effect.is_conditional():
        results.append(f'when [ {_convert_anml_timing(timing)} ] {converter.convert(effect.condition)}\n{{')
```